plugins/modules/ilo_redfish_info.py

```diff
@@ -37,79 +38,158 @@
+    required: false
     description:
       - Password for authenticating to iLO.
     type: str
   auth_token:
+    required: false
     description:
       - Security token for authenticating to iLO.
     type: str
+    version_added: 2.3.0
   timeout:
+    required: false
     description:
-      - Timeout in seconds for HTTP requests to iLO.
-    default: 10
+      - Timeout in seconds for URL requests to iLO.
+      - The default value for this param is 10 but that is being deprecated
+        and it will be replaced with 60 in community.general 5.7.0.
     type: int
+  cert_file:
+    required: false
+    description:
+      - absolute path to the server cert file
+    type: str
+  key_file:
+    required: false
+    description:
+      - absolute path to the server key file
+    type: str
 author:
-    - "Bhavya B (@bhavya06)"
+  - Bhavya B (@bhavya06)
+  - Gayathiri Devi Ramasamy (@Gayathirideviramasamy)
+  - T S Kushal (@TSKushal)
+  - Varni H P (@varini-hp)
+  - Prativa Nayak (@prativa-n)
 '''
 
 EXAMPLES = '''
   - name: Get iLO Sessions
     community.general.ilo_redfish_info:
       category: Sessions
       command: GetiLOSessions
-      baseuri: "{{ baseuri }}"
-      username: "{{ username }}"
-      password: "{{ password }}"
-    register: result_sessions
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get physical drive details
+    ilo_redfish_info:
+      category: Systems
+      command: GetPhysicalDrives
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get logical drive details
+    ilo_redfish_info:
+      category: Systems
+      command: GetLogicalDrives
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get SNMP alert destinations
+    ilo_redfish_info:
+      category: Managers
+      command: GetSnmpAlertDestinations
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get SNMP V3 Users
+    ilo_redfish_info:
+      category: Managers
+      command: GetSnmpV3Users
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get network boot settings
+    ilo_redfish_info:
+      category: Systems
+      command: GetBootSettings
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get service bios attributes
+    ilo_redfish_info:
+      category: Systems
+      command: GetServiceBiosAttributes
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get logical drive details with array controllers
+    ilo_redfish_info:
+      category: Systems
+      command: GetLogicalDrivesWithArrayControllers
+      baseuri: "***.***.***.***"
+      username: "abcxyz"
+      password: "******"
+
+  - name: Get NVMe drives details
+    ilo_redfish_info:
+      category: Systems
+      command: GetNvmeDrives
```

Comment:
Isn't this covered by "GetDiskInventory"? One of the properties returned should indicate if the drive is an NVMe drive (or some other drive type).

Reply:
We noticed that this may be redundant, removing it for now. Will revisit this in the future. Thanks.

Reply:
Resolving this conversation, thanks!